--- a/src/widgets/lottie/lv_lottie.c
+++ b/src/widgets/lottie/lv_lottie.c
@@ -134,7 +134,9 @@

     float f_total;
     tvg_animation_get_total_frame(lottie->tvg_anim, &f_total);
-    lv_anim_set_duration(lottie->anim, (int32_t)f_total * 1000 / 60); /*60 FPS*/
+    float duration = 0;
+    tvg_animation_get_duration(lottie->tvg_anim, &duration);
+    lv_anim_set_duration(lottie->anim, (uint32_t)(duration * 1000)); /*At the authored frame rate*/
     lottie->anim->act_time = 0;
     lottie->anim->end_value = (int32_t)f_total;
     lottie->anim->reverse_play_in_progress = false;
@@ -152,7 +154,9 @@

     float f_total;
     tvg_animation_get_total_frame(lottie->tvg_anim, &f_total);
-    lv_anim_set_duration(lottie->anim, (int32_t)f_total * 1000 / 60); /*60 FPS*/
+    float duration = 0;
+    tvg_animation_get_duration(lottie->tvg_anim, &duration);
+    lv_anim_set_duration(lottie->anim, (uint32_t)(duration * 1000)); /*At the authored frame rate*/
     lottie->anim->act_time = 0;
     lottie->anim->end_value = (int32_t)f_total;
     lottie->anim->reverse_play_in_progress = false;
